src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp

```diff
@@ -51,22 +74,53 @@ namespace energyplus {
       translateAndMapModelObject(lifeCycleCost);
     }
 
-    return boost::none;
-
-    /*
-  IdfObject idfObject(openstudio::IddObjectType::CommentOnly);
-
-  idfObject.setComment(
-"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\
-! OS:Space, " + modelObject.name().get() + "\n\
-!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
-
-  m_idfObjects.push_back(idfObject);
-
-  return idfObject;
-  */
-  }
+    if (m_excludeSpaceTranslation) {
+      return boost::none;
+    } else {
+      // Space
+      IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::Space, modelObject);
+
+      if (boost::optional<ThermalZone> thermalZone = modelObject.thermalZone()) {
+        idfObject.setString(SpaceFields::ZoneName, thermalZone->name().get());
+      }
+
+      idfObject.setDouble(SpaceFields::FloorArea, modelObject.floorArea());
+
+      if (boost::optional<SpaceType> spaceType = modelObject.spaceType()) {
+        idfObject.setString(SpaceFields::SpaceType, spaceType->name().get());
+      }
+
+      // Translate all Space-specific loads (and geometry children)
+      auto translateSpaceLoads = [this](auto loads) {
```

Comment:
oh my auto.

Reply:
maybe there should be a method `SpaceLoadInstance Space::loads` that returns all of the loads and then this is less needed.

Reply:
I think you'll still need the lambda somewhere for the space infiltration stuff, which has to reference a zone and not a space/spacelist.

I have a doubt about whether that actually already exists. I'll check